src/Template/AddParametersTrait.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Template;
+
+trait AddParametersTrait
+{
+    /**
+     * @var array
+     */
+    protected $templateParams = [];
+
+    /**
+     * Add parameters to template
+     *
+     * If no template name is given, the parameters will be added to all templates rendered
+     *
+     * @param array|object $params
+     * @param string|null $name
+     */
+    public function addParameters($params, $name = null)
```

Comment:
This method also needs to be added to the interface; otherwise, an implementation could omit it, leaving developers depending on concretions instead of abstractions.